internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -0,0 +1,267 @@
+	id := parse.NewFrontDoorSecurityPolicyID(profileId.SubscriptionId, profileId.ResourceGroup, profileId.ProfileName, securityPolicyName)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.SecurityPolicyName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_cdn_frontdoor_security_policy", id.ID())
+		}
+	}
+
+	profileClient := meta.(*clients.Client).Cdn.FrontDoorProfileClient
+	profile, err := profileClient.Get(ctx, profileId.ResourceGroup, profileId.ProfileName)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve the %q from the linked %q: %+v", "sku_name", "azurerm_cdn_frontdoor_profile", err)
+	}
+
+	if profile.Sku == nil {
+		return fmt.Errorf("retreving the parent %q: `sku` was nil", *profileId)
+	}
+
+	isStandardSku := strings.HasPrefix(strings.ToLower(string(profile.Sku.Name)), "standard")
+	params := cdn.BasicSecurityPolicyPropertiesParameters(nil)
+
+	if secPol, ok := d.GetOk("security_policies"); ok {
+		params, err = cdnfrontdoorsecurityparams.ExpandCdnFrontdoorFirewallPolicyParameters(secPol.([]interface{}), isStandardSku)
+		if err != nil {
+			return fmt.Errorf("expanding %q: %+v", "security_policies", err)
+		}
+	}
+
+	props := cdn.SecurityPolicy{
+		SecurityPolicyProperties: &cdn.SecurityPolicyProperties{
+			Parameters: params,
+		},
+	}
+
+	future, err := client.Create(ctx, id.ResourceGroup, id.ProfileName, id.SecurityPolicyName, props)
+	if err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for the creation of %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+	return resourceCdnFrontdoorSecurityPolicyRead(d, meta)
+}
+
+func resourceCdnFrontdoorSecurityPolicyRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Cdn.FrontDoorSecurityPoliciesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.FrontDoorSecurityPolicyID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.SecurityPolicyName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on Frontdoor Security Policy %q (Resource Group %q): %+v", id.SecurityPolicyName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.SecurityPolicyName)
+	d.Set("cdn_frontdoor_profile_id", parse.NewFrontDoorProfileID(id.SubscriptionId, id.ResourceGroup, id.ProfileName).ID())
+
+	if props := resp.SecurityPolicyProperties; props != nil {
+		securityPolicy, err := cdnfrontdoorsecurityparams.FlattenCdnFrontdoorFirewallPolicyParameters(props.Parameters)
```

Comment:
this method isn't used elsewhere, can we inline this?

Reply:
Fixed.